### DISCUSSION TITLE
Reorder Todo items?

### DISCUSSION THREAD

**Discussion by giddie** [TriliumNext/Trilium#3172] [Q&A] [answered]
I love the simplicity of CKEditor's "todo" list style to drop a quick checklist into a note. However, although the checkboxes are great, I can't find any simple way to reorder items in the list. I guess this applies to lists in general, actually. Selecting and dragging a line doesn't work great - the checkbox is left behind, and the text is instead inserted into whatever line I drop on.

This seems like a pretty important part of lists to me. Am I missing something obvious?

**Comment by zadam** [comment, marked as answer]
Yeah, that would be neat, but this is a CKEditor functionality and would have to be implemented there.

**Reply by englut** [reply]
I'd really like this feature as well.


I've made a feature request on CKEditor: https://github.com/ckeditor/ckeditor5/issues/19176

Hopefully we get this feature eventually.

**Reply by eliandoran** [reply]
@englut , you can already reorder to do items using <kbd>Alt</kbd>+<kbd>Up</kbd> and <kbd>Alt</kbd>+<kbd>Down</kbd>.
Raising the issue upstream is a good idea, since it would provide a more visual way to reorder them.

**Reply by englut** [reply]
@eliandoran oh wow. I didn't know about that. A visual way is needed for mobile however, as I'm not sure how I'd reorder the checkbox items there. Also, for my use-case, reordering checkbox items on mobile is the main concern. Do you know of a way that I can do this easily on mobile?

**Reply by englut** [reply]
Also, with `Formating toolbar` set to `Floating` in Text Note settings, I see this:

<img width="143" height="199" alt="image" src="https://github.com/user-attachments/assets/9386e37d-9d41-4084-9639-7d6b6e652faa" />

Works great on Desktop, but not visible on mobile (nor does it work when toggling to Desktop mode on Mobile).

**Reply by eliandoran** [reply]
@englut , you are absolutely right about the "Block handle" feature that allows reordering checkbox items. In that case I think you should close the upstream feature request, since the functionality is already there.

Not sure how well the handle will work on mobile, but I can probably activate it for the fixed toolbar as well (including mobile).

Update: Apparently, it's not possible to have the drag handle in the fixed editor because it actually comes as a toolbar (the one we use for the floating version). See https://ckeditor.com/docs/ckeditor5/latest/getting-started/setup/toolbar.html#block-toolbar.

**Reply by englut** [reply]
Ah so it's tied directly to the floating toolbar. Would be good to get that in mobile somehow. Or maybe the upstream request is still valid. I can update it to capture the mobile need (highlighting the block vs fixed toolbar).
